retry/backoff.go

```diff
@@ -24,3 +24,21 @@ func BackoffLinearWithJitter(waitBetween time.Duration, jitterFraction float64)
 		return backoffutils.JitterUp(waitBetween, jitterFraction)
 	}
 }
+
+// BackoffExponential produces increasing intervals for each attempt.
+//
+// The scalar is multiplied times 2 raised to the current attempt. So the first
+// retry with a scalar of 100ms is 100ms, while the 5th attempt would be 3.2s.
+func BackoffExponential(scalar time.Duration) BackoffFunc {
```

Comment:
Please run `make` so that the documentation for this get generated.

Reply:
Could you also add an example into `examples_test`

Reply:
Done

Reply:
Eh, the example bit is done. I ran `make docs`, but it did not produce anything.

Reply:
Seems to be some bug in the way that documentation is generated will file an issue. Even if I change `doc.go` it does not correctly change.